merlion/models/automl/base.py

```diff
@@ -4,35 +4,61 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
 #
-from abc import ABC
+"""
+Base class/mixin for AutoML hyperparameter search.
+"""
+from abc import abstractmethod
 from collections import Iterator
+from copy import deepcopy
 from typing import Tuple, Optional, Any
 
-from merlion.models.base import ModelBase
+from merlion.models.layers import LayeredModel, LayeredModelConfig
 from merlion.models.forecast.base import ForecasterBase
 from merlion.utils import TimeSeries
+from merlion.utils.misc import AutodocABCMeta
 
 
-class LayerMixIn(ModelBase, ABC):
+class AutoMLMixIn(LayeredModel, metaclass=AutodocABCMeta):
     """
-    Base Interface for Implemented Layers
+    Base Interface for Implemented AutoML Layers
 
     This abstract class contains all of the methods that Layers should implement. Ideally, these would be generated by
     an existing mix-in.
     """
 
+    config_class = LayeredModelConfig
+
+    def train(self, train_data: TimeSeries, **kwargs):
+        train_data = self.train_pre_process(
+            train_data, require_even_sampling=self.require_even_sampling, require_univariate=self.require_univariate
+        )
+
+        candidate_thetas = self.generate_theta(train_data)
+        theta, model, train_result = self.evaluate_theta(candidate_thetas, train_data, **kwargs)
+        if model is not None:
+            self.model = model
```

Comment:
let's rename this to not confuse with `self.config.model`, since this `model` does not necessarily have to be a Merlion model.

Reply:
I believe that the API specifies that the `model` returned by `self.evaluate_theta` is actually an instance of Merlion's `ModelBase`. At least for now, this is true of all existing implementations.